src/components/WorkOrder/MobileWorkingWorkOrderView.tsx

```diff
@@ -222,53 +233,67 @@ const MobileWorkingWorkOrderView = ({ workOrderReference }: Props) => {
     )
 
     try {
-      const filesToUpload = workOrderFiles.concat(followOnFiles)
       setCloseFormValues({
         workOrderFiles: workOrderFiles,
         followOnFiles: followOnFiles,
       })
 
-      if (filesToUpload.length > 0) {
-        if (workOrderFiles.length > 0) {
-          const uploadResult = await uploadFiles(
-            workOrderFiles,
-            workOrderReference,
-            data.workOrderPhotoDescription,
-            'Closing work order',
-            setLoadingStatus
+      cwLogger.log(`Uploading | ${filesToUpload.length} files`)
+
+      const fileGroups: {
+        files: File[]
+        description: string
+        label: string
+      }[] = [
+        {
+          files: workOrderFiles,
+          description: data.workOrderPhotoDescription,
+          label: 'Closing work order',
+        },
+        {
+          files: followOnFiles,
+          description: data.followOnPhotoDescription,
+          label: 'Raising follow-on',
+        },
+      ]
+      for (const group of fileGroups) {
+        if (group.files.length == 0) continue
+
+        const uploadResult = await uploadFiles(
+          group.files,
+          workOrderReference,
+          group.description,
+          group.label,
+          setLoadingStatus,
+          cwLogger
+        )
+
+        if (!uploadResult.success) {
+          setError(uploadResult.requestError)
+          setLoadingStatus(null)
+          const fileSummary = group.files
+            .map((f: File) => `${f.name} (${Math.round(f.size / 1000)} KB)`)
+            .join(', ')
+          cwLogger.error(
+            `Upload Error | ${filesToUpload.length} files | ${fileSummary} | ${uploadResult.requestError}`
           )
-
-          if (!uploadResult.success) {
-            setError(uploadResult.requestError)
-            setLoadingStatus(null)
-            return
-          }
-        }
-
-        if (followOnFiles.length > 0) {
-          const uploadResult = await uploadFiles(
-            followOnFiles,
-            workOrderReference,
-            data.followOnPhotoDescription,
-            'Raising a follow on',
-            setLoadingStatus
-          )
-
-          if (!uploadResult.success) {
-            setError(uploadResult.requestError)
-            setLoadingStatus(null)
-            return
-          }
+          return
         }
+        cwLogger.log(`Upload Success | ${group.files.length} files uploaded`)
       }
```

Comment:
Did a for loop here because the duplicated logic was getting to be a bit much and started to make the code hard to read